api/src/pcapi/infrastructure/repository/bank_informations/bank_informations_sql_repository.py

```diff
@@ -22,8 +23,12 @@ def get_by_application(self, application_id: int) -> BankInformations | None:
 
     def save(self, bank_informations: BankInformations) -> BankInformations:
         bank_informations_sql_entity = bank_informations_domain_converter.to_model(bank_informations)
-        repository.save(bank_informations_sql_entity)
-        bank_informations.id = bank_informations_sql_entity.id  # type: ignore [assignment]
+        # We need to do this here, because of the DDD implementation
+        # Because the instance that is going to be saved (this line upper) is not the same
+        # that the instance that is already in the session (fetch by `get_by_application` for example)
+        # id(instance_to_be_saved) != id(instance_in_the_session)
+        bank_information = db.session.merge(bank_informations_sql_entity)
+        repository.save(bank_information)
```

Comment:
Jusqu'à quand continuera-t-on à se faire du mal... ? :)

@damien-ramelet, @yacine-pc ,  @francois-seguin  : Que diriez-vous si on évaluait la complexité de remplacer toute cette cochonnerie par du code simple et clair (sans machins DDD) qui ne nécessite pas ce genre de gymnastique ? Peut-être au moins pour gérer les dossiers v5 (voire éventuellement v4, mais pas forcément v3 ou v2) ? Ça n'est pas un travail super passionnant, mais je suis assez motivé pour m'y atteler si ça ne tente personne (et que vous êtes d'accord avec l'idée, évidemment !).

Reply:
C’est dans mon death note, mais en l’état je peux pas concillier rapidité de traitement du ticket et refacto en profondeur du cette partie, j’ai déjà fait une partie en virant les forêts de ifs de la classe SaveVenueBankInformations

Reply:
`mais je ne suis quand même pas super serein` qu’est-ce qui te chagrine ? On peut se capter en meet si tu veux

Reply:
@dbaty Notre chantier comprend le traitement définitif des dossiers v2 et v3 et la suppression du code associé.
On prendra volontiers ton aide pour le nettoyage !